### PR TITLE
Doc py3

### PR DESCRIPTION
**Pull request by shashank88**
Add some starter doc for py3 migration

### CODE REVIEW

File: docs/migration_py3.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+the parameters to insert / update functions in BSONStore or wherever there is a bson.encode involved
+
+## Storing pickle in py3 and reading back in py2 fails.
+
+This could be because the default protocol for pickling in py3 is 4 which is not supported in py2 (max supported in python2 = 2).
+
+## Having str column values / index names nad column names when read back in py3 is now bytes
+
+As mentioned here: https://github.com/manahl/arctic/blob/master/arctic/serialization/numpy_records.py#L277 this can
+break the workflow of people migrating and you should be using unicode in py2 to avoid running into this or us
```
Comment (scriada) [inline docs/migration_py3.md]: `s/or us/or use/`?

File: docs/migration_py3.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+`from __future__ import unicode_literals` to always use them by default.
+
+If you hit this issue, a workaround is to set: https://github.com/manahl/arctic/blob/master/arctic/_config.py#L92
```
Comment (scriada) [inline docs/migration_py3.md]: could this link be made to mention the actual variable `FORCE_BYTES_TO_UNICODE`, incase the line number changes in future e.g.

```
[FORCE_BYTES_TO_UNICODE](https://github.com/manahl/arctic/blob/master/arctic/_config.py#L92)
```

File: docs/migration_py3.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+error message which is fixed in py3.7: https://jira.mongodb.org/browse/PYTHON-1664
+
+Arctic does not do the conversion to int from numpy.int types and you should ensure you convert it before passing
+the parameters to insert / update functions in BSONStore or wherever there is a bson.encode involved
+
+## Storing pickle in py3 and reading back in py2 fails.
+
+This could be because the default protocol for pickling in py3 is 4 which is not supported in py2 (max supported in python2 = 2).
+
+## Having str column values / index names nad column names when read back in py3 is now bytes
```
Comment (scriada) [inline docs/migration_py3.md]: I found this title a bit confusing, consider rewording? 

What about _"strings in column/index names are converted to bytes in py3"_

**Comment by shashank88** [comment]
Merging this as docs change.

**Comment by shashank88** [comment]
Merged this as docs change, and build fixed separately.

**Pull request merged**